make build-admin ERROR in ./~/admin-on-rest/lib/mui/input/RichTextInput.css

```
$ node --version
v6.9.1

$ npm --version
3.10.8

$ make build-admin
Building frontend application
Hash: aa014ab78b65d19d7bc2                                                                
Version: webpack 2.1.0-beta.25
Time: 10042ms
     Asset       Size  Chunks             Chunk Names
  index.js    9.33 MB       0  [emitted]  index
index.html  398 bytes          [emitted]  
[1131] multi index 28 bytes {0} [built]
    + 1131 hidden modules

ERROR in ./~/admin-on-rest/lib/mui/input/RichTextInput.js
Module not found: Error: Can't resolve 'quill' in '.../javascript-boilerplate/node_modules/admin-on-rest/lib/mui/input'
 @ ./~/admin-on-rest/lib/mui/input/RichTextInput.js 31:13-29
 @ ./~/admin-on-rest/lib/mui/input/index.js
 @ ./~/admin-on-rest/lib/mui/index.js
 @ ./src/admin/js/Admin.js
 @ ./src/admin/js/main.js
 @ multi index

ERROR in ./~/admin-on-rest/lib/mui/input/RichTextInput.css
Module parse failed: .../javascript-boilerplate/node_modules/admin-on-rest/lib/mui/input/RichTextInput.css Unexpected character '@' (1:0)
You may need an appropriate loader to handle this file type.
| @import "~quill/dist/quill.snow.css";
| 
| .ql-editor {
 @ ./~/admin-on-rest/lib/mui/input/RichTextInput.js 37:0-30
 @ ./~/admin-on-rest/lib/mui/input/index.js
 @ ./~/admin-on-rest/lib/mui/index.js
 @ ./src/admin/js/Admin.js
 @ ./src/admin/js/main.js
 @ multi index
Child html-webpack-plugin for "index.html":
        + 1 hidden modules
make: *** [build-admin] Error 2

```

--- a/README.md
+++ b/README.md
@@ -4,7 +4,7 @@ A starter kit for new apps, based on:
 
 * ES6 everywhere (with some bits of ES7, e.g. spread operator on objects)
 * React.js, React Router, and Redux (for the frontend)
-* Angular.js and ng-admin (for the admin)
+* Admin-on-rest (for the admin)
 * Node.js, Koa.js and PostgreSQL (for the API server)
 * Makefile, webpack and Mocha (for the build and test)
 
@@ -25,7 +25,7 @@ Features:
 * SASS preprocessor (using `node-sass`)
 * Including a non-trivial example with several routes, Ajax calls, and functional tests
 * Fully automated build with `webpack`, including development (`webpack-dev-server`) and production target (minified)
-* Admin app built with Angular and `ng-admin`
+* Admin app built with React.js and `admin-on-rest`
 * Including a full-featured admin panel with references
 * Unified test build, running unit and functional tests on all apps, powered by `mocha` and `selenium`
 * AWS deployment automated by Fabric
@@ -72,7 +72,7 @@ The entire code (api, admin, and frontend) is written in ES6 and transpiled to E
 
 **Tip**: In production, the compiled JS and CSS files (under `build/`) are served by the Node.js server. In development, it's done by webpack-dev-server.
 
-The main entry point for understanding the code is probably `src/api/server.js`.
+The main entry point for understanding the code is probably `src/api/index.js`.
 
 ## Project Configuration
 
@@ -97,8 +97,8 @@ Before running the app in development, you must copy the `config/development-dis
 - [src/api](src/api)
 - [src/frontend](src/frontend)
 - [src/admin](src/admin)
-- [e2e/api](e2e/api)
-- [e2e/frontend](e2e/frontend)
+- [src/api/e2e](src/api/e2e)
+- [src/frontend/e2e](src/frontend/e2e)
 
 ## Develop
 
@@ -118,8 +118,8 @@ make restart-frontend
 
 Browse the app:
 
-* [http://localhost:8080/admin](http://localhost:8080/admin) for the admin app
-* [http://localhost:8080/frontend](http://localhost:8080/frontend) for the frontend app
+* [http://localhost:8081](http://localhost:8081) for the admin app
+* [http://localhost:8080](http://localhost:8080) for the frontend app
 * [http://localhost:3000](http://localhost:3000) for the API
 
 **Tip**: You can change the API port by running `NODE_PORT=3001 make run-dev`. Or, for persistent change, you can add this environment variable into the [PM2 configuration file](config/pm2_servers/dev.json).
@@ -128,10 +128,6 @@ Browse the app:
 # stop servers (node and webpack)
 make stop-dev
 ```
-
-**Note:** for stability purposes dependencies versions are *exact* and production dependencies are locked down using `npm shrinkwrap --dev`. To keep fixed version update npm configuration with `npm set save-exact 1` command or append your `npm install` command with `--save-exact`.
-
-**Tip:** because versions are *exact* you can run your `npm install` faster by running `npm set cache-min Infinity` (even without internet connection if packages are already cached).
 
 ## Test
 

--- a/package.json
+++ b/package.json
@@ -39,6 +39,7 @@
     "nodemailer-html-to-text": "2.1.0",
     "nodemailer-smtp-transport": "2.7.2",
     "numeral": "1.5.3",
+    "quill": "1.1.3",
     "react": "15.3.2",
     "react-dom": "15.3.2",
     "react-fa": "4.1.2",

--- a/src/admin/index.html
+++ b/src/admin/index.html
@@ -6,7 +6,7 @@
         <meta http-equiv="X-UA-Compatible" content="IE=edge" />
         <meta name="viewport" content="width=device-width, initial-scale=1" />
     </head>
-    <body>
+    <body style="margin: 0; padding: 0;">
         <div id="root"></div>
     </body>
 </html>

--- a/src/admin/js/user/sagas.js
+++ b/src/admin/js/user/sagas.js
@@ -23,7 +23,11 @@ export const getUserFromToken = (token) => {
 
 export const signIn = (fetchSaga, storeLocalUser) => function* signInSaga({ payload: { previousRoute, ...payload } }) {
     const { error, result } = yield call(fetchSaga, { payload });
-    const user = yield call(getUserFromToken, result.token);
+
+    let user = {};
+    if (result) {
+        user = yield call(getUserFromToken, result.token);
+    }
 
     if (!error) {
         yield put(signInActions.success(user));

--- a/src/admin/js/user/sagas.js
+++ b/src/admin/js/user/sagas.js
@@ -24,12 +24,8 @@ export const getUserFromToken = (token) => {
 export const signIn = (fetchSaga, storeLocalUser) => function* signInSaga({ payload: { previousRoute, ...payload } }) {
     const { error, result } = yield call(fetchSaga, { payload });
 
-    let user = {};
-    if (result) {
-        user = yield call(getUserFromToken, result.token);
-    }
-
     if (!error) {
+        const user = yield call(getUserFromToken, result.token);
         yield put(signInActions.success(user));
         yield call(storeLocalUser, user);
         yield put(routerActions.push(previousRoute));

--- a/src/admin/webpack.config.babel.js
+++ b/src/admin/webpack.config.babel.js
@@ -72,6 +72,9 @@ export default {
         }, {
             test: /\.html$/,
             loader: 'html',
+        }, {
+            test: /\.css$/,
+            loader: 'css-loader',
         }],
     },
     output: {

